### PR TITLE
[login] module upgraded for react

### PR DESCRIPTION
**Pull request by maltheism**
### Brief summary of changes

Upgraded module to use react. I had to fix some things in LORIS to get this all to work and there are multiple changes in some JSX files as well.

What was updated to react:
Login, password expired, request account, and reset password.

Summary:
Turns out the Content-Security policy was broken for google captcha and I made a fix to include the iframe that recaptcha uses. Major was also broken inside the php/libraries/Email.class.inc and I made some fixes. That had to do with the password reset.

Note:
This PR includes the bugfix PR #4386 and so this PR can function.

### This resolves issue...

- [ ] Redmine? [#16050](https://redmine.cbrain.mcgill.ca/issues/16050)

### To test this change...

- [x] Please test on your VM.

### CODE REVIEW

**Comment by maltheism** [comment]
See #4395 for history of changes.

**Comment by kongtiaowang** [comment]
After this https://github.com/aces/Loris/pull/5222, it will pass Travis.

**Comment by johnsaigle** [comment]
This PR contains commits from other pull requests.

**Comment by johnsaigle** [comment]
You'll need to do some rewriting of this PR since #4411 has now been merged to major. Specifically a lot of `ajaxlogin.class.inc` will not work now since the Password logic has changed.

If you read over that PR you'll be able to see which parts need to be moved from the old PHP code into your new class.

**Comment by maltheism** [comment]
Hey @johnsaigle I just manually tested the PR and the functionality works without adding #4411. Maybe I can finish this PR before the overhaul needed for including the new login class.

**Comment by johnsaigle** [comment]
I think you should double-check that you are properly based on `major` because `User::isPasswordStrong()` doesn't exist anymore and your code is using it.

**Comment by johnsaigle** [comment]
>modules/login/php/ajaxlogin.class.inc:109 PhanUndeclaredVariableDim Variable $response was undeclared, but array fields are being added to it.
**modules/login/php/ajaxlogin.class.inc:192 PhanUndeclaredStaticMethod Static call to undeclared method \User::newPassword**
modules/login/php/ajaxlogin.class.inc:197 PhanUndeclaredVariableDim Variable $msg_data was undeclared, but array fields are being added to it.
**modules/login/php/ajaxlogin.class.inc:380 PhanUndeclaredStaticMethod Static call to undeclared method \User::isPasswordStrong**

Phan is telling you here that those functions don't exist anymore. You'll need to incorporate #4411.

File: modules/login/php/ajaxlogin.class.inc [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+            $values['error'] = 'Not a valid user';
+            return $values;
+        }
+
+        try {
+            // Check if valid email.
+            if ($user->isEmailValid()) {
+                $config = \NDB_Config::singleton();
+                // Generate a new password.
+                $password = \User::newPassword();
```
Comment (johnsaigle) [inline modules/login/php/ajaxlogin.class.inc]: This function doesn't exist anymore. You'll need to take a deeper look into #4411 and rework the password/login logic in this PR to use the logic from the new Password class.

**Comment by kongtiaowang** [comment]
@maltheism I will look into it.

**Comment by kongtiaowang** [comment]
@maltheism
After you fix the conflicts, I will work on your branch to modify the test cases.

**Comment by maltheism** [comment]
Close see #5327

**Pull request closed without merging**